FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90

```diff
@@ -2173,8 +2179,27 @@ subroutine atmos_phys_driver_statein (IPD_Data, Atm_block,flip_vc)
            enddo
         enddo
     endif
+    IPD_Data(nb)%Statein%dycore_hydrostatic = Atm(mytile)%flagstruct%hydrostatic
   enddo
 
  end subroutine atmos_phys_driver_statein
 
+ subroutine atmos_phys_qdt_diag(q, physics_tendency_diag, nq, dt, begin)
+   integer, intent(in) :: nq
+   real, intent(in) :: q(isd:ied,jsd:jed,1:npz,1:nq)
+   real, intent(in) :: dt
+   logical, intent(in) :: begin
+   type(physics_tendency_diag_type), intent(inout) :: physics_tendency_diag
+
+   integer :: sphum
+
+   sphum = get_tracer_index (MODEL_ATMOS, 'sphum')
+
+   if (begin) then
+      if (allocated(physics_tendency_diag%qv_dt)) physics_tendency_diag%qv_dt = q(isc:iec,jsc:jec,1:npz,sphum)
+   else
+      if (allocated(physics_tendency_diag%qv_dt)) physics_tendency_diag%qv_dt = (q(isc:iec,jsc:jec,1:npz,sphum) - physics_tendency_diag%qv_dt) / dt
+   endif
```

Comment:
This makes a lot of sense, as long as you don't have a run where only 1 timestep happens, you'd just know to not pay attention to the tendency, because it wouldn't make sense anyways

Reply:
I think this should still work in the case of a single timestep, since this routine will be called twice (once before the tracers are updated by the physics and once after).